### PR TITLE
final clean up of modal_aero_convproc.F90

### PR DESCRIPTION
**Pull request by tangshuaiqi**
This is a final clean up of modal_aero_convproc.F90 file, after refactoring the individual subroutines in it. I made the following changes:
1. re-organize the order of subroutines
2. refactor and rename a few subroutines to make more logical sense
3. move all "use" statements to the beginning of the module
4. move all parameters that only used once into the subroutine using them.
5. rename all state%xx and ptend%xx in subroutines except interface.
6. rename some variables

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 1]
```diff
@@ -454,21 +441,16 @@ subroutine ma_convproc_dp_intr(                &
...
 
-   use physics_types,  only: physics_state
-   use constituents,   only: pcnst
-
-
  
 ! Arguments
-   type(physics_state), intent(in ) :: state          ! Physics state variables
+   integer,  intent(in)    :: lchnk                       ! chunk identifier.
+   real(r8), intent(in)    :: state_pdeldry(pcols,pver)   ! layer delta-p-dry [mb]
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:451]: This is a nice solution for names of variables that reside in the `state`.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 2]
```diff
@@ -1519,6 +1507,9 @@ subroutine initialize_tmr_array(     ncnst,                  &
    real(r8) :: c_above          ! const at the above (kk-1 level) [kg/kg]
    real(r8) :: c_below          ! const at the below (kk level) [kg/kg]
 
+   real(r8), parameter :: small_con = 1.e-36        ! threshold of constitute as zero [kg/kg]
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:1514]: add `_r8`

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 3]
```diff
@@ -1920,6 +1794,9 @@ subroutine compute_activation_tend(                          &
 
    logical      :: do_act_this_lev      ! flag for doing activation at current level
    integer      :: kp1                  ! kk + 1
+   real(r8), parameter :: clw_cut = 1.0e-6      ! cutoff value of cloud water for doing updraft [kg/kg]
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: Needs `_r8`?

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 4]
```diff
@@ -2007,6 +2110,10 @@ subroutine compute_wetdep_tend(                              &
    real(r8)     :: expcdtm1     ! exp(cdt) - 1 [unitless]
    real(r8)     :: half_cld     ! 0.5 * cldfrac [fraction]
 
+   real(r8), parameter :: clw_cut = 1.0e-6      ! cutoff value of cloud water for doing updraft [kg/kg]
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: add `_r8`

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 5]
```diff
@@ -2168,11 +2248,10 @@ subroutine compute_fluxes_tendencies(                        &
...
             else
-               fluxin  =     mu_i(kp1)*conu(icnst,kp1) - ( md_i(kk )*cond(icnst,kk ) )
-               fluxout =     mu_i(kk )*conu(icnst,kk ) - ( md_i(kp1)*cond(icnst,kp1) )
-
                ! new method -- simple upstream method for the env subsidence
                ! tmpa = net env mass flux (positive up) at top of layer k
+               fluxin  =     mu_i(kp1)*conu(icnst,kp1) - ( md_i(kk )*cond(icnst,kk ) )
+               fluxout =     mu_i(kk )*conu(icnst,kk ) - ( md_i(kp1)*cond(icnst,kp1) )
                tmpa = -( mu_i(kk  ) + md_i(kk  ) )
                if (tmpa <= 0.0_r8) then
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2254]: There are some `0.0` values without `_r8` at the end lurking around. Do we care about these? Zero is a "special" floating point number in that it has an exact representation, so we could be more lenient with this particular value.

**Review by jeff-cohere — APPROVED** [review]
Very nice careful work! The formatting makes things very easy to read. I left a few minor comments.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 2 above)
Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:1514]: Adding `_r8` sometime breaks the BFBness. I think we can leave it as is as the the original code also uses this number without `_r8`.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 3 above)
Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: Same here. This number is also used without `_r8` in the original code.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 4 above)
Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: `clw_cut` is defined twice in two separate subroutines. Can we make it a module variable as it is parameter?

**Review by singhbalwinder — APPROVED** [review]
Very nice cleanup! I left some very minor comments.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 4 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: moved it into module parameter

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 2 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:1514]: as commented by Balwinder, I will keep the numbers as it as the original code

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 3 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90]: as commented by Balwinder, I will keep the numbers as it as the original code

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 5 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2254]: as commented by Balwinder, I will keep the numbers as it as the original code

Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2254]: I didn't see this comment earlier. I agree with @jeff-cohere that zero is a special number and its representation matters in single vs. double precision. I think it is better to stick with the original code implementation as any change in these numbers will break our tests which we are relying on for refactoring.

**Comment by singhbalwinder** [comment]
Thanks, @tangshuaiqi for addressing the comments. If you haven't run the tests, please do so and merge this PR.

**Comment by tangshuaiqi** [comment]
comp_testall_convproc_reclean_mam4_org_v2_baselines: 8 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**